src/domain/opendatafabric/src/dtos/dtos_extra.rs

```diff
@@ -303,3 +314,24 @@ impl Display for ExecuteQueryResponseInternalError {
         Ok(())
     }
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// SetDataSchema
+////////////////////////////////////////////////////////////////////////////////
+
+impl SetDataSchema {
+    #[cfg(feature = "arrow")]
+    pub fn new(schema: &arrow::datatypes::Schema) -> Self {
+        let (mut buf, head) = arrow::ipc::convert::schema_to_fb(schema).collapse();
+        buf.drain(0..head);
+        Self { schema: buf }
+    }
+
+    #[cfg(feature = "arrow")]
+    pub fn schema_as_arrow(&self) -> Result<arrow::datatypes::SchemaRef, crate::serde::Error> {
```

Comment:
To provide the ability to work with ODF protocol without linking to `arrow` we store schema as a byte buffer and decode on demand.